### PR TITLE
detailed error message for UserAlreadyExistsException

### PR DESCRIPTION
**Pull request by NimishaGarg**

### CODE REVIEW

File: src/main/java/dbaccessor/user/UserAccessor.java [hunk 1]
```diff
@@ -41,18 +42,34 @@ public void insertGivenUserId(User user) {
...
+        int emailIdBit = 1;
+        int setBits = 0;
+        for(User user: resultSet) {
+            if(user.getUsername().equals(username)) {
+                setBits |= (1 << usernameBit);
+            }
+            if(user.getEmailID().equals(emailID)) {
+                setBits |= (1 << emailIdBit);
+            }
+        }
```
Comment (AgamAgarwal) [inline src/main/java/dbaccessor/user/UserAccessor.java:71]: This though is efficient space wise it is hard to read and error prone.

Consider making an enum and return an `EnumSet` instead for errors.
Small enum sets are implemented as bit sets so the efficiency is maintained while the code becomes much more readable.

Comment (NimishaGarg) [reply src/main/java/dbaccessor/user/UserAccessor.java]: Done.

File: src/main/java/entity/User.java [hunk 2]
```diff
@@ -32,6 +32,10 @@
   @Column(name = "Picture")
   private String Picture;
 
+  public enum  UniqueFields {
```
Comment (AgamAgarwal) [inline src/main/java/entity/User.java:35]: Nit: double spaces.

Comment (NimishaGarg) [reply src/main/java/entity/User.java]: removed.

**Pull request merged**